server/src/test/java/org/opensearch/action/search/AbstractSearchAsyncActionTests.java

```diff
@@ -528,6 +550,544 @@ public void onFailure(Exception e) {
         assertThat(searchResponse.getSuccessfulShards(), equalTo(shards.length));
     }
 
+    public void testSearchRequestListeners() throws InterruptedException {
+        AtomicInteger dfsPreQueryPhaseStart = new AtomicInteger();
+        AtomicInteger dfsPreQueryPhaseFailure = new AtomicInteger();
+        AtomicInteger dfsPreQueryPhaseEnd = new AtomicInteger();
+        AtomicInteger canMatchPhaseStart = new AtomicInteger();
+        AtomicInteger canMatchPhaseFailure = new AtomicInteger();
+        AtomicInteger canMatchPhaseEnd = new AtomicInteger();
+        AtomicInteger queryPhaseStart = new AtomicInteger();
+        AtomicInteger queryPhaseFailure = new AtomicInteger();
+        AtomicInteger queryPhaseEnd = new AtomicInteger();
+        AtomicInteger fetchPhaseStart = new AtomicInteger();
+        AtomicInteger fetchPhaseFailure = new AtomicInteger();
+        AtomicInteger fetchPhaseEnd = new AtomicInteger();
+        AtomicInteger expandPhaseStart = new AtomicInteger();
+        AtomicInteger expandPhaseFailure = new AtomicInteger();
+        AtomicInteger expandPhaseEnd = new AtomicInteger();
+        AtomicInteger timeInNanos = new AtomicInteger(randomIntBetween(0, 10));
+        SearchRequestOperationsListener testListener = new SearchRequestOperationsListener() {
```

Comment:
Move this to private function, seems like it reused in below functions as well. Hard to read this otherwise.